### PR TITLE
[OSDOCS-3445] Amends GCP Workload Identity docs in 4.10 branch

### PR DESCRIPTION
**Pull request by jeana-redhat**
For [OSDOCS-3445](https://issues.redhat.com/browse/OSDOCS-3445)

**Previews:**
- Added **IMPORTANT** admonition to [Using manual mode with GCP Workload Identity](https://deploy-preview-44062--osdocs.netlify.app/openshift-enterprise/latest/authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.html)
- Added new procedure [Configuring the image registry to use long-lived credentials](https://deploy-preview-44062--osdocs.netlify.app/openshift-enterprise/latest/authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.html#cco-ccoctl-gcp-image-registry_cco-mode-gcp-workload-identity)
- Added **IMPORTANT** admonition to [Support for Google Cloud Platform Workload Identity](https://deploy-preview-44062--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-10-release-notes.html#ocp-4-10-auth-gcp-workload-identity) 4.10 announcement
- Added _**Support for Google Cloud Platform Workload Identity**_ to [Deprecated and removed features tracker](https://deploy-preview-44062--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-10-release-notes.html#ocp-4-10-deprecated-removed-features) with link to explanation
- Added [Image registry support for Google Cloud Platform Workload Identity removed](https://deploy-preview-44062--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-10-release-notes.html#ocp-4-10-removed-feature-auth-gcp-workload-identity) to **_Removed features_** section
- Added [short version of removal notice](https://deploy-preview-44062--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-10-release-notes.html#ocp-4-10-8-removed-feature-azure-mint-mode) to the 4.10.8 Rel Notes section

### CODE REVIEW

File: release_notes/ocp-4-10-release-notes.adoc [hunk 1]
```diff
@@ -1540,6 +1548,28 @@ $ az ad app delete --id 038c2538-7c40-49f5-abe5-f59c59c29244
...
+{product-title} 4.10.3 introduced the ability to configure the Cloud Credential Operator (CCO) to use Google Cloud Platform (GCP) Workload Identity by using the CCO utility `ccoctl`. Due to the discovery of an adverse impact to the Image Registry, this feature is removed in {product-title} 4.10.7.
+
+[IMPORTANT]
+====
+Do not configure new installations of {product-title} on GCP to use this feature.
+====
+
+For previously installed GCP clusters that use Workload Identity, upgrading to {product-title} 4.10.7 causes the Image Registry to become degraded.
+
+The configuration of Workload Identity using the `ccoctl` utility requires the CCO to operate in manual mode. To mitigate the impact to clusters that use Workload Identity, the following changes are introduced in {product-title} 4.10.7 for all {product-title} 4.10 clusters on GCP that have the CCO configured to use manual mode:
```
Comment (kalexand-rh) [inline release_notes/ocp-4-10-release-notes.adoc:1563]:
```suggestion
The configuration of Workload Identity by using the `ccoctl` utility requires the CCO to operate in manual mode. To mitigate the impact to clusters that use Workload Identity, the following changes are introduced in {product-title} 4.10.7 for all {product-title} 4.10 clusters on GCP that have the CCO configured to use manual mode:
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 2]
```diff
@@ -1540,6 +1548,28 @@ $ az ad app delete --id 038c2538-7c40-49f5-abe5-f59c59c29244
...
+
+* {product-title} 4.10 clusters on GCP that have the CCO configured to use manual mode will not be presented with updates in the web console.
+
+* Administrators who use the `oc adm upgrade` command on {product-title} 4.10 clusters on GCP that have the CCO configured to use manual mode are presented with a warning explaining the risk of updating to {product-title} 4.10.7 or later. Administrators can choose to apply the update despite the risk.
```
Comment (kalexand-rh) [inline release_notes/ocp-4-10-release-notes.adoc:1567]: This is really long. Can we break it up?

> Although it is not recommended, {product-title} 4.10 clusters on GCP that have the CCO configured to use manual mode can be updated by using the `oc` CLI. Administrators who use the `oc adm upgrade` command are presented with a warning explaining the risk of updating to {product-title} 4.10.7 or later. Administrators can choose to apply the update despite the risk.

File: authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc [hunk 3]
```diff
@@ -81,6 +81,13 @@ data:
 <3> The resource URL of the service account that can be impersonated with these credentials.
 <4> The path to the service account token inside the pod. By convention, this is `/var/run/secrets/openshift/serviceaccount/token` for {product-title} components.
 
+[IMPORTANT]
+====
+When using manual mode with GCP Workload Identity, the {product-title} image registry experiences degraded performance. In {product-title} 4.10.8, support for using GCP Workload Identity for the image registry is removed from {product-title} 4.10.
```
Comment (sdodson) [inline authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc:86]: I think we can get rid of the first sentence here. That's what we're fixing in 4.10.8, the slow performance, and that fix leads to the rest of the stuff we're talking about. Maybe consider bringing the two remaining sentences in this admonition together / cleaning them up?

File: modules/cco-ccoctl-gcp-image-registry.adoc [hunk 4]
```diff
@@ -0,0 +1,51 @@
+// Module included in the following assemblies:
+//
+// * authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc
+
+:_content-type: PROCEDURE
+[id="cco-ccoctl-gcp-image-registry_{context}"]
+= Configuring the image registry to use long-lived credentials
+
+When using manual mode with GCP Workload Identity, the {product-title} image registry experiences degraded performance. In {product-title} 4.10.8, support for using GCP Workload Identity for the image registry is removed from {product-title} 4.10.
```
Comment (sdodson) [inline modules/cco-ccoctl-gcp-image-registry.adoc:9]:
```suggestion
In {product-title} 4.10.8, support for using GCP Workload Identity for the image registry is removed from {product-title} 4.10.
```

likewise merge w/ the next sentence below?

File: release_notes/ocp-4-10-release-notes.adoc [hunk 5]
```diff
@@ -1208,7 +1208,16 @@ Users also have the option to use the `REGISTRY_AUTH_FILE` environment variable,
...
 
-//For more information, see xr/ef:../authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc#cco-mode-gcp-workload-identity[Using manual mode with GCP Workload Identity].
+For more information, see xref:../authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc#cco-mode-gcp-workload-identity[Using manual mode with GCP Workload Identity].
+
+[IMPORTANT]
+====
+When using manual mode with GCP Workload Identity, the {product-title} image registry experiences degraded performance. In {product-title} 4.10.8, support for using GCP Workload Identity for the image registry is removed from {product-title} 4.10.
+
+Starting in {product-title} 4.10.8, you must configure the image registry to use long-lived credentials. This mitigation is planned to last until Workload Identity support for the image registry is restored in a later release.
```
Comment (sdodson) [inline release_notes/ocp-4-10-release-notes.adoc:1217]: same comment here as above.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 6]
```diff
@@ -1526,6 +1540,62 @@ $ az ad app delete --id 038c2538-7c40-49f5-abe5-f59c59c29244
...
 ====
 
+[id="ocp-4-10-removed-feature-auth-gcp-workload-identity"]
+==== Image registry support for Google Cloud Platform Workload Identity removed
+
+{product-title} 4.10.3 introduced the ability to configure the Cloud Credential Operator (CCO) to use Google Cloud Platform (GCP) Workload Identity by using the CCO utility `ccoctl`.
+
+When using manual mode with GCP Workload Identity, the {product-title} image registry experiences degraded performance. In {product-title} 4.10.8, support for using GCP Workload Identity for the image registry is removed from {product-title} 4.10.
+
+Starting in {product-title} 4.10.8, you must configure the image registry to use long-lived credentials. This mitigation is planned to last until Workload Identity support for the image registry is restored in a later release.
```
Comment (sdodson) [inline release_notes/ocp-4-10-release-notes.adoc:1550]: Same here

File: authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc (hunk 3 above)
Comment (sdodson) [reply authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc:86]: And actually, I don't beleive that the performance regression needed the cluster to use Workload Identity, it's just the code changes we made for WI led to the regression which is present in all situations.

Comment (jeana-redhat) [reply authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc:86]: Ah, interesting detail :) Ok, I will revise as you suggest in all occurrences of this text :+1:

**Comment by jeana-redhat** [comment]
Updated now with @sdodson's feedback

File: modules/cco-ccoctl-gcp-image-registry.adoc [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+.Procedure
+
+. Obtain the email address of the IAM service account that the `ccoctl` created for the {product-title} image registry by running the following command:
++
+[source,terminal]
+----
+$ gcloud iam service-accounts list --filter="displayName=<name>-openshift-image-registry-gcs"
+----
++
+where `<name>` is the user-defined name for all created GCP resources used for tracking.
```
Comment (akhil-rane) [inline modules/cco-ccoctl-gcp-image-registry.adoc:28]:
```suggestion
where `<name>` is the user-defined name for all ccoctl created GCP resources used for tracking.
```

File: modules/cco-ccoctl-gcp-image-registry.adoc [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+. Create a long-lived credentials key file for the {product-title} image registry by running the following command:
++
+[source,terminal]
+----
+$ gcloud iam service-accounts keys create <path_to_image_registry_service_account_keyfile> --iam-account=<image_registry_service_account_email>
+----
++
+where:
++
+* `<path_to_image_registry_service_account_key_file>` is the path to a location you choose in which to store the image registry key file.
```
Comment (akhil-rane) [inline modules/cco-ccoctl-gcp-image-registry.adoc:39]:
```suggestion
* `<path_to_image_registry_service_account_key_file>` is the path to a location you choose in which to store the key file for the image registry's IAM service account.
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 9]
```diff
@@ -1526,6 +1538,60 @@ $ az ad app delete --id 038c2538-7c40-49f5-abe5-f59c59c29244
...
+.Procedure
+
+. Obtain the email address of the IAM service account that the `ccoctl` created for the {product-title} image registry by running the following command:
++
+[source,terminal]
+----
+$ gcloud iam service-accounts list --filter="displayName=<name>-openshift-image-registry-gcs"
+----
++
+where `<name>` is the user-defined name for all created GCP resources used for tracking.
```
Comment (akhil-rane) [inline release_notes/ocp-4-10-release-notes.adoc:1565]:
```suggestion
where `<name>` is the user-defined name for all ccoctl created GCP resources used for tracking.
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 10]
```diff
@@ -1526,6 +1538,60 @@ $ az ad app delete --id 038c2538-7c40-49f5-abe5-f59c59c29244
...
+. Create a long-lived credentials key file for the {product-title} image registry by running the following command:
++
+[source,terminal]
+----
+$ gcloud iam service-accounts keys create <path_to_image_registry_service_account_keyfile> --iam-account=<image_registry_service_account_email>
+----
++
+where:
++
+* `<path_to_image_registry_service_account_key_file>` is the path to a location you choose in which to store the image registry key file.
```
Comment (akhil-rane) [inline release_notes/ocp-4-10-release-notes.adoc:1576]:
```suggestion
* `<path_to_image_registry_service_account_key_file>` is the path to a location you choose in which to store the key file for the image registry's IAM service account.
```

**Comment by jeana-redhat** [comment]
Updated now with @akhil-rane's feedback

**Review by akhil-rane — COMMENTED** [review]
LGTM

**Comment by jeana-redhat** [comment]
Update: added the rest of the rel notes content for this release from @JoeAldinger's #44202

**Comment by jeana-redhat** [comment]
**Please note**: in the file `ocp-4-10-release-notes.adoc`, lines 2713-2724 and 2730-<end of file> are just the normal 4.10.8 release notes content, and unrelated to this feature update. 

**This PR has gone through SME review and needs to following approvals:**

Regular QE: @jianping-shu 

**Change management**

Eng: Akhil Rane @akhil-rane 
PM: Ju Lim @julienlim
Product experience: Eric Rich @sferich888 
QE: Xiaoli Tian @xltian 
Docs: Stephanie Stout @sjstout

File: authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc [hunk 11]
```diff
@@ -81,6 +81,11 @@ data:
 <3> The resource URL of the service account that can be impersonated with these credentials.
 <4> The path to the service account token inside the pod. By convention, this is `/var/run/secrets/openshift/serviceaccount/token` for {product-title} components.
 
+[IMPORTANT]
+====
+In {product-title} 4.10.8, support for using GCP Workload Identity for the image registry is removed from {product-title} 4.10, and you must configure the image registry to use long-lived credentials. This mitigation is planned to last until Workload Identity support for the image registry is restored in a later release.
```
Comment (julienlim) [inline authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc:86]: I might suggest rewording this to (as it was a bit confusing and it would be nice if we could give a reason why):

In {product-title} 4.10.8, image registry support is removed when you use GCP Workload Identity [due to ???]. To use the image registry in {product-title} 4.10.8, you must configure the image registry to use long-lived credentials instead. This mitigation is planned to last until Workload Identity support for the image registry is restored in a later release.

File: modules/cco-ccoctl-gcp-image-registry.adoc [hunk 12]
```diff
@@ -0,0 +1,49 @@
+// Module included in the following assemblies:
+//
+// * authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc
+
+:_content-type: PROCEDURE
+[id="cco-ccoctl-gcp-image-registry_{context}"]
+= Configuring the image registry to use long-lived credentials
+
+In {product-title} 4.10.8, support for using GCP Workload Identity for the image registry is removed from {product-title} 4.10, and you must configure the image registry to use long-lived credentials. This mitigation is planned to last until Workload Identity support for the image registry is restored in a later release.
```
Comment (julienlim) [inline modules/cco-ccoctl-gcp-image-registry.adoc:9]: I might suggest rewording this to (as it was a bit confusing and it would be nice if we could give a reason why):

In {product-title} 4.10.8, image registry support is removed when you use GCP Workload Identity [due to ???]. To use the image registry in {product-title} 4.10.8, you must configure the image registry to use long-lived credentials instead. This mitigation is planned to last until Workload Identity support for the image registry is restored in a later release.

Comment (jeana-redhat) [reply modules/cco-ccoctl-gcp-image-registry.adoc:9]: Thanks, Ju! How do you feel about using this bit from the rel notes?

> due to the discovery of [an adverse impact to the image registry](https://bugzilla.redhat.com/show_bug.cgi?id=2069807).

File: release_notes/ocp-4-10-release-notes.adoc [hunk 13]
```diff
@@ -1208,7 +1208,14 @@ Users also have the option to use the `REGISTRY_AUTH_FILE` environment variable,
 
 You can now use the Cloud Credential Operator (CCO) utility `ccoctl` to configure the CCO to use the Google Cloud Platform Workload Identity. When the CCO is configured to use GCP Workload Identity, the in-cluster components can impersonate IAM service accounts using short-term, limited-privilege security credentials to components.
 
-//For more information, see xr/ef:../authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc#cco-mode-gcp-workload-identity[Using manual mode with GCP Workload Identity].
+For more information, see xref:../authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc#cco-mode-gcp-workload-identity[Using manual mode with GCP Workload Identity].
+
+[IMPORTANT]
+====
+In {product-title} 4.10.8, support for using GCP Workload Identity for the image registry is removed from {product-title} 4.10, and you must configure the image registry to use long-lived credentials. This mitigation is planned to last until Workload Identity support for the image registry is restored in a later release.
```
Comment (julienlim) [inline release_notes/ocp-4-10-release-notes.adoc:1215]: I might suggest rewording this to and split this into 2 sentences (as it was a bit confusing and it would be nice if we could give a reason why):

In {product-title} 4.10.8, image registry support is removed when you use GCP Workload Identity [due to ???]. To use the image registry in {product-title} 4.10.8, you must configure the image registry to use long-lived credentials instead. This mitigation is planned to last until Workload Identity support for the image registry is restored in a later release.

**Comment by julienlim** [comment]
@jeana-redhat It mostly looks good though I found some wording a bit confusing and have suggested some changes.

File: authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc (hunk 11 above)
Comment (jeana-redhat) [reply authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc:86]: Tweaked a tiny bit, will push the update shortly. Thanks!

**Comment by jeana-redhat** [comment]
Updated now with @julienlim's feedback

**Comment by sjstout** [comment]
LGTM! Thanks, @jeana-redhat

**Comment by julienlim** [comment]
LGTM.  Thank @jeana-redhat

**Comment by sdodson** [comment]
Looks good, thanks

**Comment by jianping-shu** [comment]
Looks good to me. LGTM

File: release_notes/ocp-4-10-release-notes.adoc [hunk 14]
```diff
@@ -2644,3 +2710,39 @@ This update contains changes from Kubernetes 1.23.3 up to 1.23.5. More informati
 ==== Updating
 
 To update an existing {product-title} 4.10 cluster to this latest release, see xref:../updating/updating-cluster-cli.adoc[Updating a cluster within a minor version by using the CLI] for instructions.
+
+[id="ocp-4-10-8"]
+=== RHBA-2022:1161 - {product-title} 4.10.8 bug fix and security update
```
Comment (opayne1) [inline release_notes/ocp-4-10-release-notes.adoc:2715]:
```suggestion
=== RHSA-2022:1161 - {product-title} 4.10.8 bug fix and security update
```

Comment (opayne1) [inline release_notes/ocp-4-10-release-notes.adoc:2715]:
```suggestion
=== RHSA-2022:1162 - {product-title} 4.10.8 bug fix and security update
```

File: release_notes/ocp-4-10-release-notes.adoc [hunk 15]
```diff
@@ -2644,3 +2710,39 @@ This update contains changes from Kubernetes 1.23.3 up to 1.23.5. More informati
...
+
+Issued: 2022-04-07
+
+{product-title} release 4.10.8, which includes security updates, is now available. The bug fixes that are included in the update are listed in the link:https://access.redhat.com/errata/RHBA-2022:1161[RHBA-2022:1161] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2022:1162[RHSA-2022:1162] advisory.
```
Comment (opayne1) [inline release_notes/ocp-4-10-release-notes.adoc:2719]:
```suggestion
{product-title} release 4.10.8, which includes security updates, is now available. The bug fixes that are included in the update are listed in the link:https://access.redhat.com/errata/RHSA-2022:1162[RHSA-2022:1162] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2022:1161[RHBA-2022:1161] advisory.
```

**Review by opayne1 — COMMENTED** [review]
Sorry for the drive-by review!

File: release_notes/ocp-4-10-release-notes.adoc [hunk 16]
```diff
@@ -1208,7 +1208,14 @@ Users also have the option to use the `REGISTRY_AUTH_FILE` environment variable,
...
 
-//For more information, see xr/ef:../authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc#cco-mode-gcp-workload-identity[Using manual mode with GCP Workload Identity].
+For more information, see xref:../authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc#cco-mode-gcp-workload-identity[Using manual mode with GCP Workload Identity].
+
+[IMPORTANT]
+====
+In {product-title} 4.10.8, image registry support for using GCP Workload Identity is removed due to the discovery of link:https://bugzilla.redhat.com/show_bug.cgi?id=2069807[an adverse impact to the image registry]. To use the image registry on an {product-title} 4.10.8 that uses Workload Identity, you must configure the image registry to use long-lived credentials instead. This mitigation is planned to last until Workload Identity support for the image registry is restored in a later release.
```
Comment (nalhadef) [inline release_notes/ocp-4-10-release-notes.adoc:1215]: That pesky second sentence again.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 17]
```diff
@@ -2644,3 +2710,39 @@ This update contains changes from Kubernetes 1.23.3 up to 1.23.5. More informati
...
+
+[id="ocp-4-10-8-known-issues"]
+==== Known issues
+
+* Currently, the web console does not display virtual machine templates that are deployed to a custom namespace. Only templates deployed to the default namespace are displayed in the web console. As a workaround, avoid deploying templates to a custom namespace. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2054650[*BZ#2054650*])
+
+[id="ocp-4-10-8-bug-fixes"]
+==== Bug fixes
+* Previously, the Infrastructure Operator could not provision X11 and X12 based systems due to validation errors created by the bare metal controller (BMC) when special characters such as question marks or equals signs were used in the `filename` parameters of URLs. With this update, the `filename` parameter is removed from the URL if the virtual media image is backed by a local file. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2011626[*BZ#2011626*])
```
Comment (nalhadef) [inline release_notes/ocp-4-10-release-notes.adoc:2737]: - X11 and X12 should be hyphenated with based ("X11- and X12-based").
- Is it "equals signs" or "equal signs"?

**Review by nalhadef — COMMENTED** [review]
GCP and IAM need to be spelled out on their first use in each file.

My comments are nittish, so I'll mark this as LGTM.

File: release_notes/ocp-4-10-release-notes.adoc [hunk 18]
```diff
@@ -1208,7 +1208,14 @@ Users also have the option to use the `REGISTRY_AUTH_FILE` environment variable,
...
 
-//For more information, see xr/ef:../authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc#cco-mode-gcp-workload-identity[Using manual mode with GCP Workload Identity].
+For more information, see xref:../authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc#cco-mode-gcp-workload-identity[Using manual mode with GCP Workload Identity].
+
+[IMPORTANT]
+====
+In {product-title} 4.10.8, image registry support for using GCP Workload Identity is removed due to the discovery of link:https://bugzilla.redhat.com/show_bug.cgi?id=2069807[an adverse impact to the image registry]. To use the image registry on an {product-title} 4.10.8 that uses Workload Identity, you must configure the image registry to use long-lived credentials instead. This mitigation is planned to last until Workload Identity support for the image registry is restored in a later release.
+
+For more information, see xref:../release_notes/ocp-4-10-release-notes.adoc#ocp-4-10-removed-feature-auth-gcp-workload-identity[Image registry support for Google Cloud Platform Workload Identity removed]
```
Comment (bergerhoffer) [inline release_notes/ocp-4-10-release-notes.adoc:1217]: Missing period

File: release_notes/ocp-4-10-release-notes.adoc [hunk 19]
```diff
@@ -1395,6 +1402,11 @@ In the table, features are marked with the following statuses:
 |
 |DEP
 
+|Support for Google Cloud Platform Workload Identity
+|-
+|-
+|xref:../release_notes/ocp-4-10-release-notes.adoc#ocp-4-10-removed-feature-auth-gcp-workload-identity[REM]
```
Comment (bergerhoffer) [inline release_notes/ocp-4-10-release-notes.adoc:1408]: This is the only one in the table that has a link, just checking that it's okay to have it

File: release_notes/ocp-4-10-release-notes.adoc [hunk 20]
```diff
@@ -2644,3 +2710,39 @@ This update contains changes from Kubernetes 1.23.3 up to 1.23.5. More informati
 ==== Updating
 
 To update an existing {product-title} 4.10 cluster to this latest release, see xref:../updating/updating-cluster-cli.adoc[Updating a cluster within a minor version by using the CLI] for instructions.
+
+[id="ocp-4-10-8"]
+=== RHSA-2022:1162 - {product-title} 4.10.8 bug fix and security update
+
+Issued: 2022-04-07
+
+{product-title} release 4.10.8, which includes security updates, is now available. The bug fixes that are included in the update are listed in the link:https://access.redhat.com/errata/RHSA-2022:1162[RHSA-2022:1162] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2022:1161[RHBA-2022:1161] advisory.
```
Comment (bergerhoffer) [inline release_notes/ocp-4-10-release-notes.adoc:2719]: Links 404, which I believe is expected, but still have to mention it :)

File: release_notes/ocp-4-10-release-notes.adoc [hunk 21]
```diff
@@ -2644,3 +2710,39 @@ This update contains changes from Kubernetes 1.23.3 up to 1.23.5. More informati
...
+
+* Previously, when cloning a virtual machine from a template, changes the Operator made reverted after dismissing the dialog box if the boot disk was edited and the storage class was changed. With this update, changes made to storage class remain set after closing the dialogue box. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2049762[*BZ#2049762*])
+
+* Previously, the `startupProbe` field was added to container’s definition. As a result, `startupProbe` causes problems when creating a debug pod. With this update, `startupProbe` is removed by default from the debug pod by the `Expose --keep-startup flag` parameter, which is now set to false by default.(link:https://bugzilla.redhat.com/show_bug.cgi?id=2068474[*BZ#2068474*])
```
Comment (bergerhoffer) [inline release_notes/ocp-4-10-release-notes.adoc:2741]: * Should this say "to a container's definition"?
* Needs a space between the last period and the bug link parenthesis

File: release_notes/ocp-4-10-release-notes.adoc [hunk 22]
```diff
@@ -2644,3 +2710,39 @@ This update contains changes from Kubernetes 1.23.3 up to 1.23.5. More informati
...
+
+* Before this update, the Local Storage Operator (LSO) added an `OwnerReference` object to the persistent volumes (PV) it created, which sometimes caused an issue where a delete request for a PV could leave the PV in the `terminating` state while still attached to the pod. With this update, the LSO no longer creates an `OwnerReference` object and cluster administrators are now able to manually delete any unused PVs after a node is removed from the cluster. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2065714[*BZ#2065714*])
```
Comment (bergerhoffer) [inline release_notes/ocp-4-10-release-notes.adoc:2743]: Should "Before this update" be "Previously" to be consistent with what we normally say?

File: authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc [hunk 23]
```diff
@@ -81,6 +81,11 @@ data:
 <3> The resource URL of the service account that can be impersonated with these credentials.
 <4> The path to the service account token inside the pod. By convention, this is `/var/run/secrets/openshift/serviceaccount/token` for {product-title} components.
 
+[IMPORTANT]
+====
+In {product-title} 4.10.8, image registry support for using GCP Workload Identity is removed due to the discovery of link:https://bugzilla.redhat.com/show_bug.cgi?id=2069807[an adverse impact to the image registry]. To use the image registry on an {product-title} 4.10.8 that uses Workload Identity, you must configure the image registry to use long-lived credentials instead. This mitigation is planned to last until Workload Identity support for the image registry is restored in a later release.
```
[the remark this answers is not in the document]
Comment (bergerhoffer) [reply authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc:86]: +1

[the remark this answers is not in the document]
Comment (jeana-redhat) [reply authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.adoc:86]: Oops, yeah "cluster" got deleted at some point. Thanks!

File: release_notes/ocp-4-10-release-notes.adoc (hunk 19 above)
Comment (jeana-redhat) [reply release_notes/ocp-4-10-release-notes.adoc:1408]: Yeah, we did this for another z-stream removal recently to make sure it was clear when the removal occurred (since the table is intended more for GA than a z-stream)

File: release_notes/ocp-4-10-release-notes.adoc [hunk 24]
```diff
@@ -1526,6 +1538,60 @@ $ az ad app delete --id 038c2538-7c40-49f5-abe5-f59c59c29244
...
+
+.Prerequisites
+
+You must have:
+
+* Installed an {product-title} that uses manual mode with GCP Workload Identity.
+
+* Obtained and initialized the link:https://cloud.google.com/sdk/docs/initializing[gcloud CLI].
+
+.Procedure
```
[the remark this answers is not in the document]
Comment (jeana-redhat) [reply release_notes/ocp-4-10-release-notes.adoc:1556]: It's not the same procedure, though most of it is similar - the other procedure is for fresh installs. This procedure is only for clusters with this feature updating to 4.10.8 or later. I considered putting it in the upgrade section, but that only covers minor version upgrades, not z-streams. 

We have run into this before with a z-stream removal that requires special steps for impacted clusters... I agree that the rel notes are not really a great home for this, but we don't have a place to put things like this at the moment. I do think it's worth figuring out, but probably after this ships :)

**Review by jeana-redhat — COMMENTED** [review]
Thanks @bergerhoffer and @nalhadef!

**Comment by mffiedler** [comment]
@jianping-shu @xiuwang Please review or include other team members on the review.   Thanks.

**Review by mffiedler — APPROVED** [review]
LGTM

**Comment by jeana-redhat** [comment]
> @jianping-shu @xiuwang Please review or include other team members on the review. Thanks.

Jianping & XiuJuan, you can disregard this - Jianping has already acked. Sorry for the dropped communication with the back-and-forth here.

**Comment by jeana-redhat** [comment]
Update: peer review feedback

File: modules/cco-ccoctl-gcp-image-registry.adoc [hunk 25]
```diff
@@ -0,0 +1,47 @@
...
+. Create a long-lived credentials key file for the {product-title} image registry by running the following command:
++
+[source,terminal]
+----
+$ gcloud iam service-accounts keys create <path_to_image_registry_service_account_keyfile> --iam-account=<image_registry_service_account_email>
+----
++
+where:
++
+* `<path_to_image_registry_service_account_key_file>` is the path to a location you choose in which to store the key file for the image registry's IAM service account.
```
Comment (xiuwang) [inline modules/cco-ccoctl-gcp-image-registry.adoc:37]: s/path_to_image_registry_service_account_key_file/path_to_image_registry_service_account_keyfile

File: release_notes/ocp-4-10-release-notes.adoc [hunk 26]
```diff
@@ -1526,6 +1538,58 @@ $ az ad app delete --id 038c2538-7c40-49f5-abe5-f59c59c29244
...
+. Create a long-lived credentials key file for the {product-title} image registry by running the following command:
++
+[source,terminal]
+----
+$ gcloud iam service-accounts keys create <path_to_image_registry_service_account_keyfile> --iam-account=<image_registry_service_account_email>
+----
++
+where:
++
+* `<path_to_image_registry_service_account_key_file>` is the path to a location you choose in which to store the key file for the image registry's IAM service account.
```
Comment (xiuwang) [inline release_notes/ocp-4-10-release-notes.adoc:1574]: s/<path_to_image_registry_service_account_key_file>/<path_to_image_registry_service_account_keyfile>

**Comment by sagidlow** [comment]
PR https://github.com/openshift/openshift-docs/pull/44323 has been merged. Closing this PR in favor of the new one.
